cmd/minikube/cmd/start.go

```diff
@@ -477,16 +489,57 @@ func showVersionInfo(k8sVersion string, cr cruntime.Manager) {
 	}
 }
 
+/**
+Function to check for kubectl. The checking is to compare
+the version reported by both the client and server. It is
+that kubectl will be of the same version or 1 lower than
+the kubernetes version.
+*/
 func showKubectlConnectInfo(kcs *kubeconfig.Settings) {
+	var output []byte
+	clientVersion := kubectlversion{}
+	serverVersion := kubectlversion{}
+
+	path, err := exec.LookPath("kubectl")
+
+	if err != nil {
+		out.T(out.Tip, "For best results, install kubectl: https://kubernetes.io/docs/tasks/tools/install-kubectl/")
+	} else {
+		glog.Infof("Kubectl found at the following location %s. Let's execute and get the result", path)
+		output, err = exec.Command(path, "version", "--output=json").Output()
+
+		// ...once we get something back do some version checking
+		if err == nil {
+			glog.Infof("Received output from kubectl %s", output)
+
+			// unmarshal both the {client}
+			clientjsonErr := json.Unmarshal(output, &clientVersion)
+
+			// .......  and {server} json
+			serverjsonErr := json.Unmarshal(output, &serverVersion)
```

Comment:
No need to unmarshal the same output twice: the server version is already in `clientVersion.SVersion`